src/Platform/Platform.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Utopia\Platform;
+
+use Utopia\App;
+use Utopia\CLI\CLI;
+
+abstract class Platform
+{
+    protected array $services = [
+        'all' => [],
+        Service::TYPE_CLI => [],
+        Service::TYPE_HTTP => [],
+        Service::TYPE_GRAPHQL => []
+    ];
+
+    protected CLI $cli;
+
+    /**
+     * Initialize Application
+     *
+     * @return void
+     */
+    public function init(string $type = null): void
+    {
+        switch ($type) {
```

Comment:
@eldadfux It seems we can't seem to initialize HTTP, and CLI together, CLI seems to throw error that it only works on command line. To that respect what if we define Type in Platform as well so that a platform only accepts one type of services and inits the service based on the type set on the platform.

Reply:
Yeh, make sense, we can keep the implementation as is.